src/app/components/core/north/add-task-wizard/add-task-wizard.component.ts

```diff
@@ -305,7 +327,16 @@ export class AddTaskWizardComponent implements OnInit, OnDestroy {
             const name = payload.name
             this.uploadScript(name, files);
           }
-          this.router.navigate(['/north']);
+          if (this.source === 'flowEditor') {
+            this.ngProgress.start();
+            setTimeout(() => {
```

Comment:
This setTimeout could be removed now as we have handled API response with fork join on node editor component